unlock-app/src/components/content/MembersContent.tsx

```diff
@@ -38,18 +45,41 @@ export const MembersContent = ({ account, network, lockAddresses }: Props) => {
       {account && (
         <BrowserOnly>
           <Account network={network} account={account} />
-          <GraphWrapper lockAddresses={lockAddresses} />
+          <GraphWrapper
+            lockAddresses={lockAddresses}
+            signMetadataRequest={signMetadataRequest}
+            accountAddress={account.address}
+          />
         </BrowserOnly>
       )}
     </Layout>
   )
 }
 
-const GraphWrapper = ({ lockAddresses }: { lockAddresses: string[] }) => {
+interface GraphWrapperProps {
+  lockAddresses: string[]
+  signMetadataRequest: typeof signMetadataRequest
+  accountAddress: string
+}
+const GraphWrapper = ({
+  lockAddresses,
+  signMetadataRequest,
+  accountAddress,
+}: GraphWrapperProps) => {
   const { loading, error, data } = useQuery(keyHolderQuery(), {
     variables: { addresses: lockAddresses },
   })
 
+  useEffect(() => {
+    // Dispatch request for key metadata here, only when data changes
+    if (data) {
```

Comment:
I think this works. I also thought we might be able to query locksmith from here (instead of thru Redux) provided that we got the signature thru Redux (but don't change your approach for this for now)

Reply:
My only concern about that is the signatures are transient, so putting them in the store adds complexity because we'd also need to be able to tell when they are too stale. I think generating them at the point of use works a little simpler, although it does require the actions to carry additional data so that everything is in place for storageMiddleware.